typed-racket-doc/typed-racket/scribblings/reference/types.scrbl

```diff
@@ -351,10 +351,42 @@ corresponding to @racket[trest], where @racket[bound]
 @ex[(lambda: ([x : Any]) (if (mpair? x) x (error "not an mpair!")))]
 }
 
-@defform[(Boxof t)]{A @rtech{box} of @racket[t]}
+@defform*[[(Boxof t)
+           (Boxof write-t read-t)]]{
+  The first form is a @rtech{box} of @racket[t].
+
+  The second form is a @rtech{box} that supports write operations like
+  @racket[set-box!] taking @racket[write-t], and read operations like
+  @racket[unbox] returning @racket[read-t]. For this type to make sense,
+  @racket[write-t] should be a subtype of @racket[read-t]. This form can
+  be used to restrict writing to the box, or to safely express uncertainty
+  about the values other code could put in the box.
+}
 
 @ex[(box "hello world")]
 
+@defform[(Boxof/Read t)]{
+  A @rtech{box} of @racket[t] that only supports read-only operations
+  such as @racket[unbox]. A @racket[(Boxof/Read t)] is a subtype of
+  @racket[(Boxof t)], and a @racket[(Boxof/Read a)] is a subtype of
```

Comment:
typo: `(Boxof/Read t)` is _not_ a subtype of `(Boxof t)`

Reply:
I’ll change it to say supertype.